src/tutorials/lua_basics_lightbulb.md

```diff
@@ -365,398 +390,227 @@ switchTrigger.interactedEvent:Connect(OnSwitchInteraction)
 
-Next, we need to tell the script to switch `lightIsOn` from `false` to `true`, when we turn on the light.
+Next, we need to tell the script to switch `isSwitchOn` from `false` to `true`, when we turn on the light.
 
-In the `OnSwitchInteraction()` function, type:
+At the top of the `OnTriggerInteracted()` function, type:
 
 ```lua
-lightIsOn = not lightIsOn
+isSwitchOn = not isSwitchOn
 ```
 
-Instead of just setting `lightIsOn` to `true`, this tells the script to change `lightIsOn` to whatever it is NOT set to.
+Instead of just setting `isSwitchOn` to `true`, this tells the script to change `isSwitchOn` to the opposite of what it is already set to.
 
-If `lightIsOn` is `false` it sets it to `true`, and vice versa. In other words, the value is **toggled** to the opposite state.
+If `isSwitchOn` is `false`, it sets it to `true`, and vice versa. In other words, the value is **toggled** to the opposite state.
 
 ### Test the Variable Toggle
 
-Test to see if the script correctly toggles between `lightIsOn = false` and `lightIsOn = true` when the player interacts with the switch.
+Test to see if the script correctly toggles between `isSwitchOn = false` and `isSwitchOn = true` when the player interacts with the switch.
 
-Anywhere in our `OnSwitchInteraction` function type:
+In the `OnTriggerInteracted` function, after the `isSwitchOn = not isSwitchOn` statement, type:
 
 ```lua
-print(lightIsOn)
+print(isSwitchOn)
 ```
 
 - `print` tells the script to print to the Event Log window.
-- `lightIsOn` is the variable that will be printing.
+- `isSwitchOn` is the variable that will be printing.
 
 1. Open up the **Event Log** window from the **Window** menu in the top bar.
 2. Keep this open and press **Play**.
 3. Press ++F++ to interact with the switch.
 
+The OnTriggerInteracted function should look like:
+
+```lua
+local function OnTriggerInteracted()
+    isSwitchOn = not isSwitchOn
+    print(isSwitchOn)
+
+    Switch:RotateTo(ROTATION_ON, TIME_ROTATE, true)
+    Light.visibility = Visibility.INHERIT
+end
+```
+
 The **Event Log** should print `true` or `false` every time you interact with the light switch. You can delete the `print` statement now.
 
 ### Create an If Statement
 
 Now the script needs to know what to do specifically when the switch is on, and when it is not. An `if` statement will let you write code based on the state of other variables.
 
-In the `OnSwitchInteraction` function, after `lightIsOn = not lightIsOn` type:
+In the `OnTriggerInteracted` function, after the `lightIsOn = not lightIsOn` statement, type:
 
 ```lua
-    if not lightIsOn then
+    if isSwitchOn then
 
     end
 ```
 
 - `if` statements are handy when you need a certain series of actions to happen when a certain set of conditions is true.
-- `not lightIsOn` is the condition that must be met in order to execute the script inside our if statement.
+- `isSwitchOn` is the condition that must be met in order to execute the script inside our if statement.
```

Comment:
Do we ever explain Booleans? That a variable can hold an evaluation can be a strange concept for very new folk.